numpy/ma/core.pyi

```diff
@@ -550,7 +552,33 @@ class _extrema_operation(_MaskedUFunc):
     def reduce(self, target, axis=...): ...
     def outer(self, a, b): ...
 
-def min(obj, axis=..., out=..., fill_value=..., keepdims=...): ...
+@overload
+def min(
+    obj: _ArrayLike[_SCT],
+    axis: None = ...,
+    out: None = ...,
+    fill_value: _ScalarLike_co | None = ...,
+    keepdims: Literal[False] = ...,
+) -> _SCT: ...
+
+@overload
+def min(
+    obj: _ArrayLike[Any],
+    axis: _ShapeLike | None = ...,
+    out: None = ...,
+    fill_value: _ScalarLike_co | None = ...,
+    keepdims: bool = ...,
+) -> Any: ...
+
+@overload
+def min(
+    obj: _ArrayLike[Any],
+    axis: _ShapeLike | None = ...,
+    out: _ArrayType = ...,
+    fill_value: _ScalarLike_co | None = ...,
+    keepdims: bool = ...,
+) -> _ArrayType: ...
+
 def max(obj, axis=..., out=..., fill_value=..., keepdims=...): ...
 def ptp(obj, axis=..., out=..., fill_value=..., keepdims=...): ...
```

Comment:
`max` and `ptp` have identical signatures, so if you're not too attached to being DRY, then some good old fashioned auto-plagiarism (a.k.a. ctrl+c ctrl+v) might be an easy way to triple your winnings.

Reply:
sure, ok if i do that as a follow-up?